spec/unit/configurer_spec.rb

```diff
@@ -199,6 +199,17 @@
       @agent.run
     end
 
+    it "should create report with passed transaction_uuid and job_id" do
+      @agent = Puppet::Configurer.new(Puppet::Configurer::DownloaderFactory.new, "test_tuuid", "test_jid")
+      @agent.stubs(:init_storage)
+
+      report = Puppet::Transaction::Report.new('apply', nil, "test", "aaaa")
+      Puppet::Transaction::Report.expects(:new).with(anything, nil, anything, 'test_tuuid', 'test_jid').returns(report)
+      @agent.expects(:send_report).with(report)
+
+      @agent.run
+    end
+
```

Comment:
I think we also want a spec that the job_id is part of indirection requests for the catalog (there's a similar test for transaction_uuid).